### PR TITLE
Extract field components

### PR DESCRIPTION
**Pull request by PPiedel**
It's linked to #450

### CODE REVIEW

File: ui/client/actions/notificationActions.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
 import React from "react"
 import Notifications from 'react-notification-system-redux'
 import InlinedSvgs from "../assets/icons/InlinedSvgs"
-import Notification from "../components/notifications/Notification";
+import Notification from "../components/notifications/Notification"
+import uuid from "react-tabs/lib/helpers/uuid"
```
Comment (witekw) [inline ui/client/actions/notificationActions.js]: In other places we use - import {v4 as uuid4} from "uuid";

File: ui/client/components/graph/node-modal/editors/field/LabeledTextarea.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import ValidationLabels from "../../../../modals/ValidationLabels"
+
+export const LabeledTextarea = (props) => {
+  const {autoFocus, value, onChange, className, isMarked, readOnly, rows = 1, cols = 50, renderFieldLabel, showValidation, validators} = props
+
+  return (
+    <div className="node-row">
+      {renderFieldLabel()}
+      <div className={"node-value" + (isMarked ? " marked" : "")}>
+                <textarea
```
Comment (lciolecki) [inline ui/client/components/graph/node-modal/editors/field/LabeledTextarea.js]: Formatting - to many spaces.

File: ui/client/components/graph/node-modal/NodeDetailsContent.js [hunk 3]
```diff
@@ -488,70 +494,17 @@ export class NodeDetailsContent extends React.Component {
...
-      default:
-        return (
-          <div key={key}>
-            Field type not known...
-          </div>
-        )
-    }
+
+    return <Field fieldType={fieldType}
+                  renderFieldLabel={() => this.renderFieldLabel(fieldLabel)}
```
[the remark this answers is not in the document]
Comment (PPiedel) [reply ui/client/components/graph/node-modal/NodeDetailsContent.js]: Well, the argument here is (), not a "label". If I would pass this.renderFieldLabel as an argument, then I would have to pass also "label" to the Field component, which is not necessary

**Comment by witekw** [comment]
I've seen no regression on Firefox.

File: ui/client/components/graph/node-modal/editors/field/Checkbox.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from "react"
+import PropTypes from "prop-types"
+
+export const Checkbox = (props) => {
+  const {renderFieldLabel, autofocus, isMarked, value, onChange, readOnly} = props
+
+  return (
+    <div className="node-row">
+      {renderFieldLabel()}
```
[the remark this answers is not in the document]
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/field/Checkbox.js]: It is defined on the upper level how to render label

File: ui/client/components/graph/node-modal/MapVariable.js [hunk 5]
```diff
@@ -1,65 +1,57 @@
...
+import PropTypes from "prop-types"
+import LabeledTextarea from "./editors/field/LabeledTextarea"
+import {v4 as uuid4} from "uuid"
+import React from "react"
+import _ from "lodash"
+import Map from "./editors/map/Map"
+import {errorValidator, notEmptyValidator} from "../../../common/Validators"
+import LabeledInput from "./editors/field/LabeledInput"
 
 const MapVariable = (props) => {
```
Comment (lciolecki) [inline ui/client/components/graph/node-modal/MapVariable.js]: I think in future we should give another name for these components with postfix Node, eq. Map VariableNode. Now we have Map, Field, etc .. It's difficult to understand that.

File: ui/client/components/graph/node-modal/NodeDetailsContent.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (lciolecki) [reply ui/client/components/graph/node-modal/NodeDetailsContent.js]: Sorry, you are right ;)

File: ui/client/components/graph/node-modal/editors/field/LabeledTextarea.js (hunk 2 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/field/LabeledTextarea.js]: Done

File: ui/client/components/graph/node-modal/MapVariable.js [hunk 6]
```diff
@@ -1,65 +1,57 @@
...
+                    value={node.id}
+                    onChange={(event) => onChange("id", event.target.value)}
+                    isMarked={isMarked("id")}
+                    readOnly={readOnly}
+                    showValidation={showValidation}
+                    validators={[notEmptyValidator, errorValidator(errors, "id")]}/>
+
+      <LabeledInput renderFieldLabel={() => renderFieldLabel("Variable Name")}
+                    value={node.varName}
+                    onChange={(event) => onChange("varName", event.target.value)}
```
[the remark this answers is not in the document]
Comment (PPiedel) [reply ui/client/components/graph/node-modal/MapVariable.js]: Done

File: ui/client/actions/notificationActions.js (hunk 1 above)
Comment (PPiedel) [reply ui/client/actions/notificationActions.js]: Done

File: ui/client/components/graph/node-modal/MapVariable.js (hunk 5 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/MapVariable.js]: Yes, we should think about that

**Pull request merged**